### DISCUSSION TITLE
Quick start for Arch Linux and more rolling-release systems?

### DISCUSSION THREAD

**Discussion by ghost** [framesurge/perseus#213] [General]
I installed rust toolchain using [rustup](https://rustup.rs), and then "pacman -S gcc llvm pkgconf" to install more packages. The [Quick start](https://github.com/framesurge/perseus#quick-start) seems incomplete for Arch Linux and some more rolling-release systems, e.g., openSUSE tumbleweed, Fedora Rawhide. I don't use Debian or CentOS, because their programs are very old. The CLI output is:
perseus serve -w
⠐ [1/4] 🔨 Generating your app...
  [1/4] 🔨 Generating your app...✅
  [2/4] 🏗️  Building your app to Wasm...❌
⠐ [3/4] 📡 Building server...

**Comment by arctic-hen7** [comment]
If you're using v0.3.x, you'll need to add the Rust wasm32 toolchain as per the introductory docs. For v0.4.x, that will be done automatically. I'll make that clearer in the quick start docs, thanks for the catch!

(If this doesn't solve your problem, would you send the output of `perseus snoop wasm-build`? That might help us debug further.)

**Reply by ghost** [reply]
Arch Linux is a lightweight system, it doesn't contain wasm-bindgen-cli by default. Arch users should manually run "cargo install wasm-bindgen-cli" to install the CLI tool of wasm-bindgen.

**Reply by arctic-hen7** [reply]
Sure, but you shouldn't need to do that. Have you run `rustup target add wasm32-unknown-unknown`? You don't need the wasm-bindgen CLI to use Perseus (you will need wasm-pack though if you're using v0.3.x).

This shouldn't have anything to do with your being on Arch (others have used Perseus on Arch without problems in the past), unless something has changed recently (which it well might have).

What version of Perseus are you running?

**Reply by ghost** [reply]
> What version of Perseus are you running?

0.4.0-beta.10. I used "cargo install wasm-bindgen-cli" to succeed building wasm, otherwise, it will fail to build wasm.

**Reply by arctic-hen7** [reply]
Hmm, seems like some of the tool downloads might be failing then, which could be an Arch issue. Could you tell me if there's a `tools` directory inside the `dist` folder in your project? If not, is there a `perseus-cli` folder somewhere in your user-level cache? (It's in `~/.cache` on Debian, no clue on Arch sorry.)

**Reply by ghost** [reply]
> Could you tell me if there's a tools directory inside the dist folder in your project?

No such directory inside "dist".
> is there a perseus-cli folder somewhere in your user-level cache?

I used --target-dir parameter for cargo to specify a folder instead of the default folder.

**Reply by arctic-hen7** [reply]
I think you're misunderstanding me. In your home folder, there should be a cache directory that apps, CLIs, etc. use to store assets. In there, there should be a folder for the Perseus CLI, and I'm asking what that has in it.

Also what command are you running exactly? I'm not sure where you're specifying `--target-dir`, but thats very possibly causing this problem, since Perseus splits out the target directories for the browser and the engine.

**Reply by ghost** [reply]
In the ~/.cache folder, there are only perseus_cli/tools .

**Reply by arctic-hen7** [reply]
Okay great! And inside `tools/`?

**Reply by ghost** [reply]
> Okay great! And inside `tools/`?

Nothing. But there's a binary file ~/.cargo/bin/perseus so I can use perseus CLI.

**Reply by arctic-hen7** [reply]
Right, there's our problem. What's your architecture? Perseus automatically installs tools that are architecture dependent, and, if it can't find any, it should throw an error, but that doesn't seem to be happening.

**Reply by ghost** [reply]
ARM64.

**Reply by arctic-hen7** [reply]
Hmm, that could be an issue I suppose. Let me check exactly how Perseus would handle that and get back to you. Either way, I'm pretty confident this is arising from your architecture, not your OS.
